collector/mongodb_collector.go

```diff
@@ -40,146 +36,80 @@ func NewMongodbCollector(opts MongodbCollectorOpts) *MongodbCollector {
 // Describe describes all mongodb's metrics.
 func (exporter *MongodbCollector) Describe(ch chan<- *prometheus.Desc) {
 	glog.Info("Describing groups")
-	session, err := connectMongo(exporter.Opts.URI)
-    if err != nil{
-		return
-    }
-	serverStatus := collector_mongos.GetServerStatus(session)
-	if serverStatus != nil {
-		serverStatus.Describe(ch)
-	}
+	session := shared.MongoSession(exporter.Opts.URI)
 	defer session.Close()
-}
-
-func connectMongo(uri string)(*mgo.Session, error) {
-        r, _ := regexp.Compile("connect=direct")
-        if r.MatchString(uri) != true {
-            r, _ := regexp.Compile("\\?")
-            if r.MatchString(uri) == true {
-                uri = uri + "&connect=direct"
-            } else {
-                uri = uri + "?connect=direct"
-            }
-        }
-	session, err := mgo.Dial(uri)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
-        os.Exit(1)
-		glog.Errorf("Cannot connect to server using url: %s", uri)
-		return nil,err
-	}
-
-	session.SetMode(mgo.Eventual, true)
```

Comment:
Both of those were moved to the 'MongoSession' func in [shared/connection.go](https://github.com/timvaillancourt/prometheus_mongodb_exporter/blob/upstream_updates_bugfixes/shared/connection.go#L15-34)